### ISSUE TITLE
No Duplicate Shadows Not Functioning as Intended

### ISSUE THREAD

**Issue by dylan-mcbee** [rotobash/pokemon-ngc-rando#35]
Found two milotics within the first couple of hours of gameplay

**Comment by rotobash** [comment]
Hi there, thanks for reporting this.

This should be easy enough to fix, let me take a look this weekend.

**Comment by dylan-mcbee** [comment]
Sounds good! I actually tried a fix on my end as well, each time a mon is randomized I populated that hashset, then in the while loop after setting the index, I see if the index is in that hashset, and if it is I set the index back to 0 to run it again. Seems to be working so far for my friend trying it

**Comment by rotobash** [comment]
Hey so I took a look, seems like I was asleep at the wheel when I implemented this option the first time. The setting was added, the hashset was even initialized, but the logic was missing 😅 

But you're exactly right, hash sets are perfect for checks like this and the algorithm you're describing is the way I would do it. I added a few safety checks in case other options conflict with picking a duplicate, feel free to take a look at the PR if you're interested.
 
I also found out my legendary list isn't correct either, this actually fixes the "No Legendaries" option as well. So thanks again for the bug report! I have a few other changes to merge before making a new release but keep an eye out for it soon. Cheers.
